_posts/2019-03-18-speed-up-your-pipeline-using-cache.md

```diff
@@ -0,0 +1,125 @@
+---
+layout: post
+title:  Speed up your pipeline using the cache
+date: 2019-03-29
+tags: codebuild aws docker cache continuous-integration
+author: JK Gunnink
+---
+
+Fast feedback in development is an important consideration when deploying software often as
+it helps reduce context switching. When a developer creates some functionality, and wants to test
+and validate the functionality using a CI tool, fast feedback is important to maintain focus.
+
+In a recent project I worked on, one of the tasks I was given was to speed up the time it took for
+the CI tools to report success or failure in order to validate the work, but also get the work
+merged in faster. The application is a containerised ruby-on-rails application which has two status
+checks which are required before any work can be merged in. They are:
+- Build the image and test
+- Build the image for production
+
+Using AWS codebuild for both checks, using containers and caching enabled a 84.2% improvement on the
+time taken to get a response. From 4m:45seconds down to 45 seconds.
+
+By leveraging caching layers of docker containers, we were able to take advantage of rebuilding the
+image fast. We started out with a buildspec file that looked like so:
+```yaml
```

Comment:
I've read through this a couple of times, and it's still not clear based on the examples/snippets.

- Introduce what caching is and how it works - enough for the reader to follow (link elsewhere for further reading if required) <possibly pulling up from `# The result` section>
- The Dockerfile is missing - just realised that's a missing piece.
- introduce how to use caching in codebuild.